### PR TITLE
Don't show icons if they're null

### LINKED ISSUE #764: Observations: null icon
### Description

Currently if there is no icon for observations, the default icon is 

Image

This can cause people to think there is a hazard in their area.

We should instead replace this with an icon (or no icon) that better communicates this error. 

### Acceptance criteria
- [ ] Tests have been implemented or modified to capture the bug
- [ ] Changes have been tested for accessibility

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

- adds a centralized thing for figuring out icons
- don't display icons if they're null

Closes #764

### CODE REVIEW

File: web/modules/weather_data/src/Service/WeatherDataService.php [hunk 1]
```diff
@@ -370,31 +360,33 @@ public function getApiObservationKey($observation)
...
-        $path = array_map(function ($piece) {
-            return preg_replace("/,.*$/", "", $piece);
-        }, $path);
+            $path = array_map(function ($piece) {
+                return preg_replace("/,.*$/", "", $piece);
+            }, $path);
 
-        $apiConditionKey = implode("/", $path);
+            $key = implode("/", $path);
+            $icon->icon = $this->legacyMapping->$key->icon;
```
Comment (eric-gade) [inline web/modules/weather_data/src/Service/WeatherDataService.php:385]: I guess in theory this is where an uncaught error could emerge (if there is no corresponding key on the mapping for some inexplicable reason). Kind of a note to myself for #837

File: web/themes/new_weather_theme/templates/block/block--weathergov-current-conditions.html.twig [hunk 2]
```diff
@@ -28,9 +28,11 @@
       </div>
 
       <div class="display-flex margin-bottom-2">
+        {% if content.icon.icon %}
         <div class="icon margin-right-105 maxw-5 maxh-5">
-          {{ source(directory ~ "/assets/images/weather/icons/conditions/" ~ content.icon )}}
+          {{ source(directory ~ "/assets/images/weather/icons/conditions/" ~ content.icon.icon )}}
         </div>
+        {% endif %}
```
Comment (eric-gade) [inline web/themes/new_weather_theme/templates/block/block--weathergov-current-conditions.html.twig:35]: We should switch to the link:href svg from the spritesheet here right?

**Review by eric-gade — APPROVED** [review]
Good refactor!
  
## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [ ] (if applicable) Post deploy steps are run
- [ ] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [ ] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

File: web/themes/new_weather_theme/templates/block/block--weathergov-current-conditions.html.twig (hunk 2 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/block/block--weathergov-current-conditions.html.twig:35]: I think so, yeah.

**Pull request merged**